crates/sui-single-node-benchmark/src/benchmark_context.rs

```diff
@@ -232,4 +233,23 @@ impl BenchmarkContext {
             *gas_objects = Arc::new(refreshed_gas_objects);
         }
     }
+
+    pub(crate) async fn validator_sign_transactions(
+        &self,
+        transactions: Vec<Transaction>,
+    ) -> Vec<HandleTransactionResponse> {
+        info!(
+            "Started signing {} transactions. You can now attach a profiler",
+            transactions.len(),
+        );
+        let tasks: FuturesUnordered<_> = transactions
+            .into_iter()
+            .map(|tx| {
+                let validator = self.validator();
+                tokio::spawn(async move { validator.sign_transaction(tx).await })
```

Comment:
there is no need to spawn a task here, since you do `tasks.collect().await`

Reply:
Actually Zhe points out that this may not run in parallel, so we may want to leave the spawn here.

Reply:
At least for tx signing benchmark, removing the spawn actually makes it slightly faster. It could be because signing is too cheap that parallelism doesn't help

Reply:
Yeah I thought about the `spawn()` pattern for the initial PRs. Maybe only for executions using a separate task makes sense.